server/scsynth/SC_World.cpp

```diff
@@ -968,9 +968,9 @@ void World_Cleanup(World* world, bool unload_plugins) {
 
 #ifndef NO_LIBSNDFILE
         if (nrtbuf->sndfile)
-            sf_close(nrtbuf->sndfile);
-        if (rtbuf->sndfile && rtbuf->sndfile != nrtbuf->sndfile)
-            sf_close(rtbuf->sndfile);
+            sf_close(GETSNDFILE(nrtbuf));
+        if (rtbuf->sndfile && rtbuf->sndfile != GETSNDFILE(nrtbuf))
```

Comment:
@Spacechild1 is this check correct now? I wasn't sure here

Reply:
rtbuf->sndfile && rtbuf->sndfile != nrtbuf->sndfile is perfectly ok.

Reply:
Thanks! Updated.